### DISCUSSION TITLE
Some issues with game operation

### DISCUSSION THREAD

**Discussion by Eduradoxu** [rohanpsingh/LearningHumanoidWalking#15] [General]
Hey!Dear Author，You have opened up an excellent project.I have almost successfully started your project,but still exist some problems.
1.After I have trained the env of 'jvrc_walk',I used ''debug_stepper" to view the effect.The result was that every time the game was restarted, the robot always moved in a different mode, such as walking forward or standing still ,etc. I can't know how to modify your code about the "walk_task".
2.And then, like the above situation, I trained the env of "jvrc_step".I still use "debug_stepper" to view.The result is that the robot directly penetrates the stairs and walks forward, as if it does not have a physical collision with the stairs. At the same time, I do not know how to make the robot able to go downstairs and walk normally, as you have shown.

Sorry, I'm still a rookie student who just got to know bipedal robots and DRL,It's still difficult for me to familiarize myself with your code.I would greatly appreciate your answer and assistance to me.Looking forward to your answer, I hope every day in the future will be full of achievements for you！

**Comment by rohanpsingh** [comment]
Hi,
>The result was that every time the game was restarted, the robot always moved in a different mode, such as walking forward or standing still ,etc.

This is expected because the walking mode is randomized even during evaluation (for eg., at the start of each episode [here](https://github.com/rohanpsingh/LearningHumanoidWalking/blob/main/tasks/stepping_task.py#L267-L269)). What do you want to do exactly?

> The result is that the robot directly penetrates the stairs and walks forward, as if it does not have a physical collision with the stairs

Can you share a short video clip or a picture to show this?

**Reply by Eduradoxu** [reply]
https://github.com/rohanpsingh/LearningHumanoidWalking/assets/73418116/39dca715-e3db-4b35-b0bc-584e9dd52caf
I am surprised that you replied to me so quickly. I have attached a video to you. For your first reply, I would like to know how to modify the code so that each play can walk in a fixed way, rather than relying on probability.For stepping_task I can see the corresponding code for the random pattern , but  how is walk_task accomplished? I can't modify it; For the second response, as shown in the video, the reason for the fall may be because I only trained 6000 iterations.

**Reply by rohanpsingh** [reply]
Hi, I can't play the video - it's all black. 
I understand your second question better now and I think it could be an issue due to the version of mujoco. Could you try after doing `pip install mujoco==2.3.0`? Make sure you're able to `play` the pre-trained policy provided in this repository without any problems, before you train your own policy from scratch.

For the first question, sorry, there's really no user-friendly way to achieve this. You have to manually edit the code. For eg., you can comment out [this line](https://github.com/rohanpsingh/LearningHumanoidWalking/blob/main/tasks/stepping_task.py#L267-L269) and set `self.mode = WalkModes.FORWARD`.

**Reply by Eduradoxu** [reply]
1. You are right. I would like to ask what your "dm-control" version is. I have installed "mujoco 2.2.0" version, which should be the problem. However, using 1.1.0 and 1.0.0 versions of "dm control" is not useful.
2. I have tried to modify the code as you mentioned, but it does not work for the "jvrc_walk" environment, as it does not have this line of code in its code. What I want to know is how to modify this environment"jvrc_walk" to achieve a fixed walking mode.

**Reply by rohanpsingh** [reply]
Hi @Eduradoxu 
Sorry for the late reply.
1. `dm_control` version does not matter. Which `mujoco` version do you currently have? Please confirm using this command: `$ python -c "import mujoco; print(mujoco.__version__)"` 
2. In `jvrc_walk.py` there are only two modes: standing still, or moving forward at a speed between 0.3 to 0.4m/s. You can hardcode this over [here](https://github.com/rohanpsingh/LearningHumanoidWalking/blob/main/tasks/walking_task.py#L77) if you want.

**Reply by Eduradoxu** [reply]
Hello，my friend!I'm glad you can still reply to me amidst your busy schedule. My "mujoco" used the version 2.3.0 you mentioned, and then I fixed the "modes" to "forward". After training 8000 times, I was excited to be able to go upstairs as you demonstrated. However, in the scene of going downstairs, I failed. The ladder was below the floor, causing the robot to only walk on it. I changed the height of the floor but it still didn't work.

**Reply by rohanpsingh** [reply]
Hi, 
I think this is also due to version of mujoco. Can you add the following lines [here](https://github.com/rohanpsingh/LearningHumanoidWalking/blob/main/tasks/stepping_task.py#L308) and retry:
```
            self._client.model.geom('floor').group = 3                                                                                  
            self._client.model.geom('floor').conaffinity = 0                                                                            
            self._client.model.geom('floor').contype = 0
```
This will disable the collisions with the "floor" and not visualize it be default.

**Comment by Eduradoxu** [comment]
Hi, with your help, I have been able to play games very well. I will continue to study your paper and code in the future, and I believe that your research can help many Chinese students like me. Thank you sincerely again for your answer !

**Reply by rohanpsingh** [reply]
Thanks, I'm glad you could solve it!
